src/renderer/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { createContext, useMemo, useState } from 'react';
+import MenuItem from '@mui/material/MenuItem';
+import { clipboard } from 'electron';
+import { useTranslation } from 'react-i18next';
+import { ContextData } from './types';
+import * as Styled from './style';
+
+interface Props {
+  children: React.ReactNode[];
+}
+
+type HandlerFunctionCreator = (
+  data: ContextData
+) => (event: React.MouseEvent) => void;
+
+export const MenuContext = createContext<HandlerFunctionCreator>(
+  () => () => {}
+);
+
+export default function ContextMenu({ children }: Props) {
+  const { t } = useTranslation();
+  const [contextMenu, setContextMenu] = useState<{
+    data: ContextData;
+    element: Element;
+    mouseX: number;
+    mouseY: number;
+  } | null>(null);
+
+  const createContextMenuHandler = useMemo(
+    () => (data?: ContextData) => (event: React.MouseEvent) => {
+      event.preventDefault();
+      event.stopPropagation();
+      setContextMenu(
+        contextMenu === null
+          ? {
+              data: data || {},
+              element: document.activeElement || event.currentTarget,
+              mouseX: event.clientX + 2,
+              mouseY: event.clientY - 6,
+            }
+          : null
+      );
+    },
+    []
+  );
```

Comment:
To memo a function it's best to use the `useCallback` instead of `useMemo`.

```suggestion
  const createContextMenuHandler = useCallback(
    (data?: ContextData) => (event: React.MouseEvent) => {
      event.preventDefault();
      event.stopPropagation();
      setContextMenu(
        contextMenu === null
          ? {
              data: data || {},
              element: document.activeElement || event.currentTarget,
              mouseX: event.clientX + 2,
              mouseY: event.clientY - 6,
            }
          : null
      );
    },
    []
  );
```